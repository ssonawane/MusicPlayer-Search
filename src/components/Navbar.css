.main {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-color: #333;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dropbtn {
  display: inline-block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

.dropdown:hover .dropbtn {
  background-color: #9e9e9e;
}

.dropdown {
  flex: 7.5;
}

.search {
  flex: 8;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
  left: 240px;
}

.dropdown-content input {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content input:hover {
  background-color: #f1f1f1;
}

.dropdown:hover .dropdown-content {
  display: block;
}

.search input {
  height: 30px;
  width: 300px;
  padding-left: 10px;
}

.dropdown-option {
  display: inherit;
}
