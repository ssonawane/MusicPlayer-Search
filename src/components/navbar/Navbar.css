.main {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-color: #333;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dropbtn {
  display: inline-block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

.dropdown:hover .dropbtn {
  background-color: #9e9e9e;
  width: 30%;
}

.dropdown {
  flex: 1;
}

.search {
  flex: 1;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 250px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
  left: 212px;
}

.dropdown-content input {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content input:hover {
  background-color: #f1f1f1;
}

.dropdown:hover .dropdown-content {
  display: block;
}

.search input {
  height: 30px;
  width: 300px;
  padding-left: 10px;
}

@media (max-width: 768px) {
  .main {
    display: flex;
    flex-direction: column;
  }

  .dropdown-content {
    width: 100%;
    left: 0px;
  }

  .search {
    margin: 8px;
  }

  .search input {
    width: 280px;
  }

  .dropdown:hover .dropbtn {
    background-color: #333333;
    width: 100%;
  }
}
