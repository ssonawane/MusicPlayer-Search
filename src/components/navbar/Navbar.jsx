import React from 'react';
import style from "./Navbar.module.css";
import SearchInput from "../searchinput/SearchInput";
import DropdownFilter from "../dropdown/DropdownFilter";

function Navbar(props) {

    const getSearchIp = (searchStr) => {
        props.getSearchIp(searchStr);
    }
    const getDrpDwnFilter = (filter) => {
        props.getDrpDwnFilter(filter);
    }

    return <div className={style.main}>
        <DropdownFilter getDrpDwnFilter={getDrpDwnFilter} />
        <SearchInput getSearchIp={getSearchIp} />
    </div>
}

export default Navbar;