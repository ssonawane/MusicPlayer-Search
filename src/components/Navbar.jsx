import React from 'react';
import "./Navbar.css";

function Navbar(props) {
    return <div className="main">
        {props.children}
    </div>
}

export default Navbar;