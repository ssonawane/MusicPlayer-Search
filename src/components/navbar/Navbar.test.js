import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Navbar from "./Navbar";

describe("Navbar  ", () => {
  test("renders learn react link", () => {
    const NavbarComponent = renderer.create(<Navbar />).toJSON();
    expect(NavbarComponent).toMatchSnapshot();
  });
});
