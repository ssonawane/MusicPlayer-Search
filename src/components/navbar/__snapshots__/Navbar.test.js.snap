// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar   renders learn react link 1`] = `
<div
  className="main"
>
  <div
    className="dropdown"
  >
    <a
      className="dropbtn"
      data-testid="activity"
      href="#"
    >
      Activities
    </a>
    <div
      className="dropdown-content"
      data-testid="drpdwn-content"
    >
      <div
        className="checkboxContainer"
      >
        <label
          className="container"
          data-testid="All"
        >
          All
          <input
            checked={false}
            data-testid="all"
            onChange={[Function]}
            type="checkbox"
            value="all"
          />
          <span
            className="checkmark"
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="search"
  >
    <form>
      <input
        data-testid="input-search"
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
